qa-physics/buildChargeTree.groovy

```diff
@@ -48,6 +57,31 @@ dataFile.eachLine { line ->
   ufcStop = tok[r++].toBigDecimal()
   livetime = tok.size()>11 ? tok[r++].toBigDecimal() : -1
 
+  // open monitor_<runnum>.hipo
+  // data_table.dat has lines that repeat the run number and bin number.
+  // Use runPrevious and binPrevious to avoid repeating the extraction of the histogram bin contents
+  if(runnum!=runPrevious){
+    runPrevious = runnum
+    monFile = new File("${monDir}/monitor_${runnum}.hipo")
+    println "Opening $monFile"
+    if(!(monFile.exists())) throw new Exception("monitor<run>.hipo not found")
+    try {
+      inMdir.readFile(monDir + "/" + monFile.getName())
+    } catch(Exception ex) {
+      System.err.println("ERROR: cannot read file $monFile; it may be corrupt")
+      return
+    }
+  }
+
+  if(binnum!=binPrevious){
```

Comment:
It hasn't happened yet, but if you have two consecutive runs which only have a single bin, then `binPrevious == 0` at the end of the first run, but `binnum` will again be `0` when we get to the second run. I think you can fix this by resetting `binPrevious = -1` when the run number changes.